app/controllers/admin/creator_settings_controller.rb

```diff
@@ -1,15 +1,26 @@
 module Admin
   class CreatorSettingsController < Admin::ApplicationController
-    ALLOWED_PARAMS = %i[community_name logo_svg primary_brand_color_hex invite_only_mode public checked_code_of_conduct
-                        checked_terms_and_conditions].freeze
+    ALLOWED_PARAMS = %i[community_name primary_brand_color_hex invite_only_mode public checked_code_of_conduct
+                        checked_terms_and_conditions logo].freeze
 
-    def new; end
+    def new
+      @max_file_size = LogoUploader::MAX_FILE_SIZE
+      @logo_allowed_types = (LogoUploader::CONTENT_TYPE_ALLOWLIST +
+        LogoUploader::EXTENSION_ALLOWLIST.map { |extension| ".#{extension}" }).join(",")
+    end
 
     def create
       extra_authorization
+
+      raise CarrierWave::IntegrityError if settings_params[:logo].blank?
+
       ActiveRecord::Base.transaction do
+        logo_uploader = upload_logo(settings_params[:logo])
+        ::Settings::General.original_logo = logo_uploader.url
+        # An SVG will not be resized, hence we apply the OR statements below to populate SETTINGS consistently.
+        ::Settings::General.resized_logo = logo_uploader.resized_logo.url || logo_uploader.url
```

Comment:
Love the comment for context, but this looks like logic that might make more sense in the model.

```ruby
::Settings::General.resized_logo = logo_uploader.resized_logo_url
```

This is not a blocker, but something that might help encapsulate logic.

Reply:
@jeremyf to clarify, is this what you were thinking: 

```
          logo_uploader = upload_logo(settings_params[:logo])
          ::Settings::General.original_logo = logo_uploader.url
          ::Settings::General.resized_logo = ::Settings::General.determine_logo_url(logo_uploader)
  ```

and in the model `models/settings.general.rb`

```
    def self.determine_logo_url logo_uploader
      # An SVG will not be resized, hence we apply the OR statements below to populate SETTINGS consistently.
      logo_uploader.resized_logo.url || logo_uploader.url
    end
```